### ISSUE TITLE
RecurringJob does not trigger callback properly

### ISSUE THREAD

**Issue by hanabi1224** [HangfireIO/Hangfire#1413]
Hi,
I recently found that RecurringJob does not trigger callback properly when the callback depends on fields/properties in class. Is this bug or by design?

**Comment by odinserj** [comment]
Hi @hanabi1224. Could you give an example?

**Comment by hanabi1224** [comment]
@odinserj  Yes here I created a repro: https://github.com/HangfireIO/Hangfire/pull/1414

**Comment by odinserj** [comment]
Sorry, I don't understand the difference between working and non-working wrapper.

**Comment by hanabi1224** [comment]
@odinserj Just added some comments in PR thread, please let me know if it does not help : )

**Comment by hanabi1224** [comment]
@odinserj  kindly ping, any updates?
